Fix a few E2E tests that may fail due to network time lag or no fallback requests

eason9487 opened the pull request:
### Changes proposed in this Pull Request:

This PR fixes a few E2E tests.
- Fix the tests of pre-launch checklist
   - It may fail due to the network time lag.
- Fix the test of claimed GMC account.
   - The multiple routes of the same pattern didn't fall back.

#### Context

While working on #2108 and #2115, there were a few tests that always failed or failed very easily.

Especially with the change to lazy loading in #2108, `page.goto( url, { waitUntil: 'domcontentloaded' } )` may lead to skipping or failing tests because the lazy loading page is not fully loaded and the target elements are not yet available.

![image](https://github.com/woocommerce/google-listings-and-ads/assets/17420811/29ca8994-13d7-4fcc-988a-f8660e8727f7)

(Ref: https://github.com/woocommerce/google-listings-and-ads/actions/runs/6337008219/job/17211193354#step:10:397)

### Detailed test instructions:

1. View the E2E test results of this PR on GHA: https://github.com/woocommerce/google-listings-and-ads/actions/runs/6337174014/job/17211701721
2. Set up local E2E test env.
3. Run `npm run test:e2e -- ./tests/e2e/specs/setup-mc/step-1-accounts.test.js`, all tests should pass.
4. Run `npm run test:e2e -- ./tests/e2e/specs/setup-mc/step-3-confirm-store-requirements.test.js`, all tests should pass.
5. Run `npm run test:e2e` to make sure all tests are passed.

### Additional details:

#### 📌 Test case of claimed GMC account:

In the following code, the last registered `setUpAccountsPage.mockMCCreateAccountWebsiteClaimed` will override `setUpAccountsPage.mockMCHasNoAccounts` as they share the same route pattern. It caused the page to crash because the first request expected an empty array of GMC accounts but it returned a claimed account data as an object structure. Therefore, the test always resulted in a timeout failure.

https://github.com/woocommerce/google-listings-and-ads/blob/47bbbdfe93beba9482d87197681f0dd8ea9de9e3/tests/e2e/specs/setup-mc/step-1-accounts.test.js#L301-L315

![1](https://github.com/woocommerce/google-listings-and-ads/assets/17420811/94253f4a-26e9-44c7-99c3-c6cb73a6ccb1)

#### 📌 Test cases of pre-launch checklist:

The [result of `prelaunchChecklistCheckboxes.all()`](https://github.com/woocommerce/google-listings-and-ads/blob/47bbbdfe93beba9482d87197681f0dd8ea9de9e3/tests/e2e/specs/setup-mc/step-3-confirm-store-requirements.test.js#L290-L292) has a very high chance of being an empty array, and it happens on the subsequent tests.

![2](https://github.com/woocommerce/google-listings-and-ads/assets/17420811/40973917-6075-4586-becb-a8b527d56f81)

### Changelog entry

eason9487 committed:
Fix the e2e test of claimed GMC account because multiple routes of the same pattern didn't fall back.

eason9487 committed:
Fix the e2e tests of pre-launch checklist, which may fail due to the network time lag.

tomalec commented on tests/e2e/specs/setup-mc/step-3-confirm-store-requirements.test.js:
```diff
@@ -284,25 +284,25 @@ test.describe( 'Confirm store requirements', () => {
 
 				test( 'should toggle all checklists and see the "Confirm" button in each checklist', async () => {
-					const prelaunchChecklistToggles =
-						storeRequirements.getPrelaunchChecklistToggles();
```
I'd remove `getPrelaunchChecklistToggles`
https://github.com/woocommerce/google-listings-and-ads/blob/dev/fix-e2e-network-issues/tests/e2e/utils/pages/setup-mc/step-3-confirm-store-requirements.js#L197:L206

Or do you envision using that function later?

eason9487 replied:
Thanks! Removed in https://github.com/woocommerce/google-listings-and-ads/pull/2116/commits/4ebc91d6c0bc91cc321e00b59927d66f0cc57920.

tomalec reviewed (approved):
I reviewed the code and ran tests locally. 
Tests are passing on every consecutive run, code LGTM.

Thank you for the detailed explanations! :pray: 

I left one note on removing dead code, but it should not affect the results.

eason9487 committed:
Remove the unused `getPrelaunchChecklistToggles` method from the `StoreRequirements` class.

Address: https://github.com/woocommerce/google-listings-and-ads/pull/2116#discussion_r1340638959
